lib/opam_solve.ml

```diff
@@ -196,6 +196,86 @@ module Opam_monorepo_context (Base_context : BASE_CONTEXT) :
     | Error _ -> false
     | Ok opam_file -> Opam.has_cross_compile_tag opam_file
 
+  (* this is a hack - as we don't have a global view of the package
+     universe, we build the dev_repo table lazily, whenever the 0install
+     solver request more information about a package (by calling
+     [Solver.candidates]. *)
+  let dev_repos = ref Dev_repo.Map.empty
+
+  let hashes pkg =
+    match OpamFile.OPAM.url pkg with
+    | None -> []
+    | Some url ->
+        (* note: pinned packages do not have any checksums set *)
+        OpamFile.URL.checksum url
+
+  (* build the list of conflicts by removing packages with the same hash *)
+  let conflicts_with_same_dev_repo_but_a_different_hash =
+    let memo = Hashtbl.create 13 in
```

Comment:
Anything particularly useful about picking 13 as initial size or is this just some number?

Reply:
That's a good small prime number to init such structures ;-)